File: python-flask/src/application/dto.py
from dataclasses import dataclass


@dataclass
class CreateOrderItemInput:
    productId: str
    quantity: int
    price: float


@dataclass
class CreateOrderInput:
    customerId: str
    items: list[CreateOrderItemInput]


@dataclass
class CreateOrderOutput:
    orderId: str
    customerId: str
    totalAmount: float
    itemsCount: int
    createdAt: str

File: nodejs-nestjs/src/application/dto.ts
import { IsString, IsNotEmpty, IsArray, ValidateNested, IsNumber, Min, ArrayMinSize } from 'class-validator';
import { Type } from 'class-transformer';

export class CreateOrderItemInput {
  @IsString()
  @IsNotEmpty()
  productId: string;

  @IsNumber()
  @Min(1)
  quantity: number;

  @IsNumber()
  @Min(0)
  price: number;
}

export class CreateOrderInput {
  @IsString()
  @IsNotEmpty()
  customerId: string;

  @IsArray()
  @ArrayMinSize(1)
  @ValidateNested({ each: true })
  @Type(() => CreateOrderItemInput)
  items: CreateOrderItemInput[];
}

export class CreateOrderOutput {
  orderId: string;
  customerId: string;
  totalAmount: number;
  itemsCount: number;
  createdAt: string;
}
